machine_learning_hep/utilities_validation.py

```diff
@@ -30,6 +30,44 @@ def fillvalidationvsmult(dfevt, dfevtevtsel, df_reco):
     Create a TH3F histogram and fill it with three variables from a dataframe.
     """
     _ = len(df_reco)
+    hlist = []
+    def makenfill(nx, xl, xh, ny, yl, yh, X, Y):
+      hn = "h_" + X + "_vs_" + Y
+      ht = " ; " + X + " ; " + Y
+      h = None
+      if "perc" in Y:
+        a = [yl]
+        while a[-1] + 0.1 < 10:
+          a.append(a[-1] + 0.1)
+        while a[-1] + 1 < 100:
+          a.append(a[-1] + 1)
+        ar = TArrayF(len(a))
+        for i,j in enumerate(a):
+          ar.SetAt(j, i)
+        h = TH2F(hn, ht, nx, xl, xh, len(a) - 1, 0, 1)
+        h.GetYaxis().Set(len(a) - 1, ar.GetArray())
```

Comment:
You don't need to create a `TArrayF` here. You can use `array('d', a)` directly in the `TH2F` constructor. You just need to import `array` like this: `from array import array`

Reply:
Yes, indeed, I will use the functions as defined in `utilities_plot.py`